server/src/main/java/org/opentosca/toscana/plugins/cloudformation/visitor/CloudFormationNodeVisitor.java

```diff
@@ -0,0 +1,277 @@
+                .engine("MySQL")
+                .dBName(dbName)
+                .masterUsername(masterUser)
+                .masterUserPassword(masterPassword)
+                .dBInstanceClass(dBInstanceClass)
+                .allocatedStorage(allocatedStorage)
+                .storageType(storageType)
+                .vPCSecurityGroups(cfnModule.fnGetAtt(securityGroupName, "GroupId"));
+        } catch (Exception e) {
+            logger.error("Error while creating DBInstance resource.");
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void visit(MysqlDbms node) {
+        logger.debug("Visit MysqlDbms node " + node.getNodeName() + ".");
+        // TODO what to do if there is a configure script
+        // Get the host of this DBMS node
+        // String host = toAlphanumerical(node.getHost().getCapability().getResourceName().get());
+    }
+
+    @Override
+    public void visit(Apache node) {
+        logger.debug("Visit Apache node " + node.getNodeName() + ".");
+        // check if host is available
+        ComputeCapability computeCapability = node.getHost().getCapability();
+        if (computeCapability.getResourceName().isPresent()) {
+            //Hosted on name
+            String host = toAlphanumerical(computeCapability.getResourceName().get());
+
+            cfnModule.getCFNInit(host)
+                .getOrAddConfig(CONFIG_SETS, CONFIG_INSTALL)
+                .putPackage(
+                    //TODO apt only if linux
+                    new CFNPackage("apt")
+                        .addPackage("apache2"));
+        }
+    }
+
+    @Override
+    public void visit(WebApplication node) {
+        logger.debug("Visit WebApplication node " + node.getNodeName() + ".");
+
+        //get the name of the server where this node is hosted on
+        String serverName;
+        if (node.getHost().getFulfillers().size() == 1) {
+            WebServer webServer = node.getHost().getFulfillers().toArray(new WebServer[1])[0];
+            serverName = toAlphanumerical(webServer.getHost().getCapability().getResourceName().get());
+        } else {
+            throw new IllegalStateException("More than one or no fulfiller");
+        }
+
+        if (node.getStandardLifecycle().getCreate().isPresent()) {
+            Operation create = node.getStandardLifecycle().getCreate().get();
+            handleOperation(create, serverName, CONFIG_INSTALL);
+        }
+
+        if (node.getStandardLifecycle().getConfigure().isPresent()) {
+            Operation configure = node.getStandardLifecycle().getConfigure().get();
+            handleOperation(configure, serverName, CONFIG_CONFIGURE);
+        }
+    }
+
+    private String checkOrDefault(Optional<String> optional, String def) {
+        return optional.isPresent() ? optional.get() : def;
+    }
+
+    private Integer checkOrDefault(Optional<Integer> optional, Integer def) {
+        return optional.isPresent() ? optional.get() : def;
```

Comment:
you could consider using `optional.orElse(def)`.

This also applies to the other occrances